src/entries/popup/pages/settings/customRpc.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useCallback, useState } from 'react';
+
+import { useCustomRPCsStore } from '~/core/state/customRPC';
+import { isValidUrl } from '~/core/utils/connectedApps';
+import { Box, Button, Inline, Stack, Text } from '~/design-system';
+import { Input } from '~/design-system/components/Input/Input';
+
+import { maskInput } from '../../components/InputMask/utils';
+
+export function SettingsNetworksCustomRPC() {
+  const { customRPCs, addCustomRPC } = useCustomRPCsStore();
+  const [customRPC, setCustomRPC] = useState<{
+    rpcUrl?: string;
+    chainId?: number;
+    name?: string;
+    symbol?: string;
+    explorerUrl?: string;
```

Comment:
also, can we add another flag "active" ? that way we know which one to use it in case you have multiple rpcs for the same chain Id

Reply:
oh right